// angular-plotly.js/esm2020/lib/plotly-shared.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PlotlyService } from './plotly.service';
import { PlotlyComponent } from './plotly.component';
import * as i0 from "@angular/core";
export class PlotlySharedModule {
    constructor() { }
}
PlotlySharedModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.6", ngImport: i0, type: PlotlySharedModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PlotlySharedModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.6", ngImport: i0, type: PlotlySharedModule, declarations: [PlotlyComponent], imports: [CommonModule], exports: [PlotlyComponent] });
PlotlySharedModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.6", ngImport: i0, type: PlotlySharedModule, providers: [PlotlyService], imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.6", ngImport: i0, type: PlotlySharedModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PlotlyComponent],
                    imports: [CommonModule],
                    providers: [PlotlyService],
                    exports: [PlotlyComponent],
                }]
        }], ctorParameters: function () { return []; } });
